src/main/java/org/dcsa/ovs/service/impl/OVSEventSubscriptionTOServiceImpl.java

```diff
@@ -0,0 +1,254 @@
+  private final Function<OVSEventSubscriptionTO, EventSubscription>
+      eventSubscriptionTOToEventSubscription =
+          esTo -> {
+            EventSubscription eventSubscription = new EventSubscription();
+            eventSubscription.setSubscriptionID(esTo.getSubscriptionID());
+            eventSubscription.setCallbackUrl(esTo.getCallbackUrl());
+            eventSubscription.setSecret(esTo.getSecret());
+            eventSubscription.setCarrierServiceCode(esTo.getCarrierServiceCode());
+            eventSubscription.setCarrierVoyageNumber(esTo.getCarrierVoyageNumber());
+            eventSubscription.setVesselIMONumber(esTo.getVesselIMONumber());
+            eventSubscription.setTransportCallID(esTo.getTransportCallID());
+            return eventSubscription;
+          };
+
+  private Mono<OVSEventSubscriptionTO> createEventTypes(
+      OVSEventSubscriptionTO eventSubscriptionTO) {
+
+    List<EventType> eventTypes;
+
+    if (CollectionUtils.isEmpty(eventSubscriptionTO.getEventType())) {
+      eventTypes = ALL_ALLOWED_EVENT_TYPES;
+      eventSubscriptionTO.setEventType(eventTypes);
+    } else {
+      eventTypes = eventSubscriptionTO.getEventType();
+    }
+
+    return Flux.fromIterable(eventTypes)
+        .concatMap(
+            eventType ->
+                eventSubscriptionRepository.insertEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), eventType))
+        .then(Mono.just(eventSubscriptionTO));
+  }
+
+  private Mono<Void> createTransportEventType(OVSEventSubscriptionTO eventSubscriptionTO) {
+    List<TransportEventTypeCode> transportEventTypeCode =
+        eventSubscriptionTO.getTransportEventTypeCode();
+    return Flux.fromIterable(transportEventTypeCode)
+        .flatMap(
+            t ->
+                eventSubscriptionRepository.insertTransportEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), t))
+        .then();
+  }
+
+  private Mono<Void> createOperationsEventType(OVSEventSubscriptionTO eventSubscriptionTO) {
+    List<OperationsEventTypeCode> operationsEventTypeCode =
+        eventSubscriptionTO.getOperationsEventTypeCode();
+    return Flux.fromIterable(operationsEventTypeCode)
+        .flatMap(
+            o ->
+                eventSubscriptionRepository.insertOperationsEventTypeForSubscription(
+                    eventSubscriptionTO.getSubscriptionID(), o))
+        .then();
+  }
+
+  @Override
+  public Mono<OVSEventSubscriptionTO> update(OVSEventSubscriptionTO eventSubscriptionTO) {
+    if (eventSubscriptionTO.getSecret() != null) {
+      return Mono.error(
+          new UpdateException(
+              "Please omit the \"secret\" attribute.  If you want to change the"
+                  + " secret, please use the dedicated secret endpoint"
+                  + " (\"PUT .../event-subscriptions/"
+                  + eventSubscriptionTO.getSubscriptionID()
+                  + "/secret\")."));
+    }
```

Comment:
Same remark as for TNT (this should be handled in the subclass somehow)

Reply:
Added hook.